tests/gateway/tools.py

```diff
@@ -95,14 +97,22 @@ def save_yaml(path, conf):
 
 class MockAsyncResponse:
     def __init__(self, data: Dict[str, Any]):
-        self.data = data
+        # Extract status and headers from data, if present
+        self.status = data.get("_status", 200)
```

Comment:
We can add more arguments to `__init__` if necessary.

Reply:
I was trying to minimize the number of tests that we'd have to rewrite in this PR